src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java

```diff
@@ -66,12 +71,24 @@
      */
     private DirectoryScanner directoryScanner;
 
+    /**
+     * Used for detecting changes between the Mojo execution.
+     * @see #getGeneratedSourcesDirectoryScanner() ;
+     */
+    private DirectoryScanner generatedSourcesDirectoryScanner;
+
     /**
      * Once the {@link #beforeRebuildExecution(org.apache.maven.shared.incremental.IncrementalBuildHelperRequest)} got
      * called, this will contain the list of files in the build directory.
      */
     private String[] filesBeforeAction = new String[0];
 
+    /**
+     * Once the {@link #beforeRebuildExecution(org.apache.maven.shared.incremental.IncrementalBuildHelperRequest)} got
```

Comment:
After the {@link #beforeRebuildExecution(org.apache.maven.shared.incremental.IncrementalBuildHelperRequest)} is
     * called, this contains the list of files in the build directory.

Reply:
ping